src/main/kotlin/reminder/domain/ReminderRepository.kt

```diff
@@ -5,4 +5,5 @@ import java.util.*
 interface ReminderRepository {
     fun save(reminder: Reminder)
     fun search(identifier: ReminderIdentifier): Optional<Reminder>
+    fun delete(identifier: ReminderIdentifier): Boolean
```

Comment:
I don't think that returning a boolean is the better choice here, since if it fails (returning false) you don't know which error is happening (is it because an I/O error? Because the reminder doesn't exist? etc).

I think that throwing exceptions for non-desirable cases is a better way to manage the "unusual" cases.

Anyway, I don't know if we should not have any return parameter or returning the deleted reminder should be a better option....

Reply:
I don't think returning the reminder itself is the best way to handle this, as it technically doesn't exist anymore. 
But I agree with the part that making it return `unit` is also not the best practice. 
Given this, what do you think it's better:
- Returning the reminder
- Returning the identifier
- Returning nothing

Reply:
I think that returning the reminder identifier isn't really needed, since you are passing it as a method parameter (so, it means that you already know it).

Returning the reminder itself means that you can do some operations with it (read the title and description about the reminder) that otherwise you couldn't do (since you have just deleted the entry inside the repository, and, as a consequence the getReminder method will return Optional.None).

But the real question here is: Do we really need to check the information of a removed reminder? For now we don't so maybe if the future leads us to this need, we can change this use case and adapt it.

This means that I would opt for returning Unit, and if the needs change, we adapt this use case for our needs.